src/main/java/com/pgs/client/service/UserClient.java

```diff
@@ -0,0 +1,82 @@
+package com.pgs.client.service;
+
+import com.pgs.client.component.Client;
+import com.pgs.client.dto.UserDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class UserClient {
+
+    private final RestTemplate restTemplate;
+
+    @Value("${app.url.user}")
+    private String apiUsersUrl;
+
+    public UserDto getSingleUser(Long id) {
+        HttpHeaders headers = setHeaders();
+        var request = new HttpEntity<>(headers);
+        return restTemplate.exchange(
+                apiUsersUrl + "/" + id,
+                HttpMethod.GET,
+                request,
+                UserDto.class).getBody();
+    }
+
+    public UserDto addUser(UserDto userDto) {
+        HttpHeaders headers = setHeaders();
+        var request = new HttpEntity<>(userDto, headers);
+        return restTemplate.exchange(
+                apiUsersUrl,
+                HttpMethod.POST,
+                request,
+                UserDto.class).getBody();
+    }
+
+    public UserDto activateUser(Long id) {
+        HttpHeaders headers = setHeaders();
+        var request = new HttpEntity<>(headers);
+        return setUserEnabled(apiUsersUrl+"/"+id+"/activate", request);
+    }
+
+    public UserDto deactivateUser(Long id) {
+        HttpHeaders headers = setHeaders();
+        var request = new HttpEntity<>(headers);
+        return setUserEnabled(apiUsersUrl+"/"+id+"/deactivate", request);
+    }
```

Comment:
There is still come common stuff in both of those methods. The private one should have a structure like so:
```java
private UserDto setUserEnabled(Long userId, boolean enabled)
```